lightkube/codecs.py

```diff
@@ -83,15 +85,24 @@ def load_all_yaml(stream: Union[str, TextIO], context: dict = None, template_env
     """
     if context is not None:
         stream = _template(stream, context=context, template_env=template_env)
-    resources = []
-    for obj in yaml.safe_load_all(stream):
-        if obj is not None:
-            res = from_dict(obj)
-            resources.append(res)
-
-            if create_resources_for_crds is True and res.kind == "CustomResourceDefinition":
-                create_resources_from_crd(res)
-    return resources
+
+    def _flatten(objects: Iterator) -> List[AnyResource]:
+        """Flatten resources which hae a kind = *List."""
+        resources = []
+        for obj in objects:
+            if obj is None:
+                continue
+            if isinstance(obj, Mapping) and obj.get("kind").endswith("List"):
```

Comment:
If you are trying to avoid an exception at this stage (a check for presence of `kind` is done later in `from_dict` ), `.get()` would need an empty string as fallback:
`obj.get("kind", "").endswith("List")`

Reply:
yeah, nice.  that's precisely what i was trying to avoid